flake8_trio.py

```diff
@@ -1290,22 +1285,75 @@ def visit_Call(self, node: ast.Call):
         self.generic_visit(node)
 
 
+class Visitor200(Flake8TrioVisitor):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.async_function = False
+
+    def visit_AsyncFunctionDef(
+        self, node: ast.AsyncFunctionDef | ast.FunctionDef | ast.Lambda
+    ):
+        outer = self.get_state("async_function")
+        self.async_function = isinstance(node, ast.AsyncFunctionDef)
+        self.generic_visit(node)
+        self.set_state(outer)
+
+    visit_FunctionDef = visit_AsyncFunctionDef
+    visit_Lambda = visit_AsyncFunctionDef
+
+    def visit_Call(self, node: ast.Call):
+        if self.async_function and (
+            key := fnmatch_qualified_name(
+                [node.func], *self.options.trio200_blocking_calls.keys()
+            )
+        ):
+            self.error("TRIO200", node, key, self.options.trio200_blocking_calls[key])
+        self.generic_visit(node)
+
+
 class ListOfIdentifiers(argparse.Action):
     def __call__(
         self,
         parser: argparse.ArgumentParser,
         namespace: argparse.Namespace,
-        values: Sequence[Any] | None,
+        values: Sequence[str] | None,
         option_string: str | None = None,
     ):
         assert values is not None
         assert option_string is not None
         for value in values:
             if not value.isidentifier():
-                raise argparse.ArgumentError(self, f"{value} is not a valid identifier")
+                raise argparse.ArgumentError(
+                    self, f"{value!r} is not a valid identifier"
```

Comment:
`if keyword.iskeyword(value) or not value.isidentifier():` and `f"{value!r} is not a valid variable name"` - we don't want people putting in `True`, for example.